packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -884,7 +884,7 @@ abstract class ResidentRunner {
       }
       return true;
     } else if (character == 'a') {
-      if (supportsServiceProtocol && isRunningDebug) {
+      if (supportsServiceProtocol && (isRunningDebug || isRunningProfile)) {
```

Comment:
this is equivalent to checking for supportsServiceProtocol

Reply:
Done.